### PR TITLE
Validate golden test configs and parse the test cases.

### PR DESCRIPTION
**Pull request by chloechien**

### CODE REVIEW

**Comment by chloechien** [comment]
Credit to @drevell  who find out fstest does not support windows :(

File: templates/commands/goldentest/test_funcs.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+	entries, err := os.ReadDir(testDir)
+	if err != nil {
+		return nil, fmt.Errorf("error reading golden test directory (%s): %w", testDir, err)
+	}
+
+	testCases := []*TestCase{}
+	for _, entry := range entries {
+		if !entry.IsDir() {
+			return nil, fmt.Errorf("unexpeted file entry under golden test directory: %s", entry.Name())
```
Comment (drevell) [inline templates/commands/goldentest/test_funcs.go:75]: "unexpeted" -> "unexpected"

**Review by drevell — APPROVED** [review]

File: templates/commands/goldentest/test_funcs.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/abcxyz/abc/templates/model/goldentest"
+)
+
+// TestCase describes a template golden test case.
+type TestCase struct {
+	// Name of the test case.
```
[the remark this answers is not in the document]
Comment (chloechien) [reply templates/commands/goldentest/test_funcs.go:28]: Done.

File: templates/commands/goldentest/test_funcs.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	// named testdata/golden.
+	goldenTestDir = "testdata/golden"
+
+	// The golden test config file is always located in the test case root dir and
+	// named test.yaml.
+	configName = "test.yaml"
+)
+
+// ParseTestCases returns a list of test cases to record or verify.
+func ParseTestCases(location, testName string) ([]*TestCase, error) {
```
[the remark this answers is not in the document]
Comment (chloechien) [reply templates/commands/goldentest/test_funcs.go:46]: Good catch! Updated.

File: templates/commands/goldentest/test_funcs.go (hunk 1 above)
Comment (chloechien) [reply templates/commands/goldentest/test_funcs.go:75]: Thank you!

File: templates/commands/render/flags.go [hunk 4]
```diff
@@ -169,16 +168,3 @@ func (r *RenderFlags) Register(set *cli.FlagSet) {
 		return nil
 	})
 }
-
```
[the remark this answers is not in the document]
Comment (chloechien) [reply templates/commands/render/flags.go:172]: :)

**Comment by chloechien** [comment]
Closed in https://github.com/abcxyz/abc/pull/221

**Pull request closed without merging**